src/main/java/com/northconcepts/datapipeline/examples/cookbook/UpsertRecordsToPostgreSql.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2006-2020 North Concepts Inc.  All rights reserved.
+ * Proprietary and Confidential.  Use is subject to license terms.
+ * 
+ * https://northconcepts.com/data-pipeline/licensing/
+ */
+package com.northconcepts.datapipeline.examples.cookbook;
+
+import java.io.File;
+import java.io.OutputStreamWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.DataWriter;
+import com.northconcepts.datapipeline.csv.CSVReader;
+import com.northconcepts.datapipeline.csv.CSVWriter;
+import com.northconcepts.datapipeline.jdbc.JdbcReader;
+import com.northconcepts.datapipeline.jdbc.JdbcUpsertWriter;
+import com.northconcepts.datapipeline.jdbc.upsert.PostgreSqlUpsert;
+import com.northconcepts.datapipeline.job.Job;
+
+public class UpsertRecordsToPostgreSql {
+    public static void main(String[] args) throws Throwable {
+        final String DATABASE_DRIVER = "org.postgresql.Driver";
+        final String DATABASE_URL = "jdbc:postgresql://localhost:5432";
+        final String DATABASE_USERNAME = "test";
+        final String DATABASE_PASSWORD = "test";
+        final String DATABASE_TABLE = "CreditBalance";
+        final String PRIMARY_KEY = "Account";
+
+        Class.forName(DATABASE_DRIVER);
+		Connection connection = DriverManager.getConnection(DATABASE_URL, DATABASE_USERNAME, DATABASE_PASSWORD);
```

Comment:
More indentation issues.

Reply:
converted indentation to spaces and fixed it.